core/src/main/scala/org/apache/gearpump/partitioner/CoLocationPartitioner.scala

```diff
@@ -0,0 +1,13 @@
+package org.apache.gearpump.partitioner
+
+import org.apache.gearpump.Message
+
+/**
+ * Will have the same parallism with last processor
+ * And each task in current processor will co-locate with task of last processor
+ */
+class CoLocationPartitioner extends Partitioner {
+  override def getPartition(msg : Message, partitionNum : Int, currentPartitionId: Int = -1) : Int = {
+    currentPartitionId
```

Comment:
1. what shall we return if currentPartitionId is -1?
2. it is better to name -1 as a CONSTANT. E.g. NO_PARTITION_ASSIGNED